test/src/unit-constructor1.cpp

```diff
@@ -813,6 +813,21 @@ TEST_CASE("constructors")
             CHECK(j.type() == json::value_t::number_float);
         }
 
+        SECTION("NaN")
+        {
+            // NaN is stored properly, but serialized to null
+            json::number_float_t n(std::numeric_limits<json::number_float_t>::quiet_NaN());
+            json j(n);
+            CHECK(j.type() == json::value_t::number_float);
+
+            // check round trip of NaN
+            json::number_float_t d = j;
+            CHECK(d == n);
```

Comment:
Apparently, comparing NaNs with `==` does not work, hence all CI builds fail. Please check https://stackoverflow.com/questions/15268864/how-to-compare-two-nan-values-in-c.

Reply:
Okay, I'll close this. Thanks for reminding : )

Reply:
I did not mean to close this - maybe

```cpp
CHECK(std::is_nan(d) and std::is_nan(n));
```

would also do the job?